### ISSUE TITLE
Install fails at installing torch.  "BadZipFile: Bad CRC-32" 

### ISSUE THREAD

**Issue by lamachine** [bmaltais/kohya_ss#822]
Had an issue with installing torch,  "BadZipFile: Bad CRC-32"  The fix was this (closed) issue   https://github.com/bmaltais/kohya_ss/issues/732.  You must be in the folder with webui-user.bat for one of the commands to find the requirements document.  For some reason, all requirements are met, but it still wants to install torch again.  I tried commenting that out in launch but no joy.

Win11, new install Py 3.10

I have looked through all of these issue, and one led me to a solution from another repository, but no joy.

I updated the requirements, installed all the dependencies directly into python, and confirmed with pip freeze against the requirements.txt.  I double click webui-user.bat and it tries to install torch again and fails.  

I really want to use the Web UI, but my next step is to see if I  can install command line version.  

<D:\Stable Diffusion\stable-diffusion-webui>git pull
Already up to date.
venv "D:\Stable Diffusion\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Version: v1.2.1
Commit hash: 89f9faa63388756314e8a1d96cf86bf5e0663045
Installing torch and torchvision
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cu118
Collecting torch==2.0.1
  Using cached https://download.pytorch.org/whl/cu118/torch-2.0.1%2Bcu118-cp310-cp310-win_amd64.whl (2619.1 MB)
Collecting torchvision==0.15.2
  Using cached https://download.pytorch.org/whl/cu118/torchvision-0.15.2%2Bcu118-cp310-cp310-win_amd64.whl (4.9 MB)
Requirement already satisfied: filelock in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from torch==2.0.1) (3.12.0)
Requirement already satisfied: jinja2 in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from torch==2.0.1) (3.1.2)
Requirement already satisfied: networkx in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from torch==2.0.1) (3.1)
Requirement already satisfied: sympy in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from torch==2.0.1) (1.12)
Requirement already satisfied: typing-extensions in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from torch==2.0.1) (4.5.0)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from torchvision==0.15.2) (9.5.0)
Requirement already satisfied: requests in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from torchvision==0.15.2) (2.30.0)
Requirement already satisfied: numpy in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from torchvision==0.15.2) (1.24.3)
Requirement already satisfied: MarkupSafe>=2.0 in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from jinja2->torch==2.0.1) (2.1.2)
Requirement already satisfied: certifi>=2017.4.17 in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from requests->torchvision==0.15.2) (2023.5.7)
Requirement already satisfied: urllib3<3,>=1.21.1 in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from requests->torchvision==0.15.2) (2.0.2)
Requirement already satisfied: charset-normalizer<4,>=2 in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from requests->torchvision==0.15.2) (3.1.0)
Requirement already satisfied: idna<4,>=2.5 in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from requests->torchvision==0.15.2) (3.4)
Requirement already satisfied: mpmath>=0.19 in d:\stable diffusion\stable-diffusion-webui\venv\lib\site-packages (from sympy->torch==2.0.1) (1.3.0)
Installing collected packages: torch, torchvision
**ERROR: Exception:
Traceback (most recent call last):
  File "D:\Stable Diffusion\stable-diffusion-webui\venv\lib\site-packages\pip\_internal\cli\base_command.py", line 160, in exc_logging_wrapper
    status = run_func(*args)
  File "D:\Stable Diffusion\stable-diffusion-webui\venv\lib\site-packages\pip\_internal\cli\req_command.py", line 247, in wrapper
    return func(self, options, args)
  File "D:\Stable Diffusion\stable-diffusion-webui\venv\lib\site-packages\pip\_internal\commands\install.py", line 507, in run
    installed = install_given_reqs(
  File "D:\Stable Diffusion\stable-diffusion-webui\venv\lib\site-packages\pip\_internal\req\__init__.py", line 73, in install_given_reqs
    requirement.install(
  File "D:\Stable Diffusion\stable-diffusion-webui\venv\lib\site-packages\pip\_internal\req\req_install.py", line 796, in install
    install_wheel(
  File "D:\Stable Diffusion\stable-diffusion-webui\venv\lib\site-packages\pip\_internal\operations\install\wheel.py", line 729, in install_wheel
    _install_wheel(
  File "D:\Stable Diffusion\stable-diffusion-webui\venv\lib\site-packages\pip\_internal\operations\install\wheel.py", line 589, in _install_wheel
    file.save()
  File "D:\Stable Diffusion\stable-diffusion-webui\venv\lib\site-packages\pip\_internal\operations\install\wheel.py", line 388, in save
    shutil.copyfileobj(f, dest)
  File "C:\Users\Owner\AppData\Local\Programs\Python\Python310\lib\shutil.py", line 195, in copyfileobj
    buf = fsrc_read(length)
  File "C:\Users\Owner\AppData\Local\Programs\Python\Python310\lib\zipfile.py", line 927, in read
    data = self._read1(n)
  File "C:\Users\Owner\AppData\Local\Programs\Python\Python310\lib\zipfile.py", line 1017, in _read1
    self._update_crc(data)
  File "C:\Users\Owner\AppData\Local\Programs\Python\Python310\lib\zipfile.py", line 945, in _update_crc
    raise BadZipFile("Bad CRC-32 for file %r" % self.name)
zipfile.BadZipFile: Bad CRC-32 for file 'torch/bin/fbgemm.dll'**

[notice] A new release of pip is available: 23.0.1 -> 23.1.2
[notice] To update, run: D:\Stable Diffusion\stable-diffusion-webui\venv\Scripts\python.exe -m pip install --upgrade pip
Traceback (most recent call last):
  File "D:\Stable Diffusion\stable-diffusion-webui\launch.py", line 369, in <module>
    prepare_environment()
  File "D:\Stable Diffusion\stable-diffusion-webui\launch.py", line 271, in prepare_environment
    run(f'"{python}" -m {torch_command}', "Installing torch and torchvision", "Couldn't install torch", live=True)
  File "D:\Stable Diffusion\stable-diffusion-webui\launch.py", line 95, in run
    raise RuntimeError(f"""{errdesc or 'Error running command'}.
RuntimeError: Couldn't install torch.
Command: "D:\Stable Diffusion\stable-diffusion-webui\venv\Scripts\python.exe" -m pip install torch==2.0.1 torchvision==0.15.2 --extra-index-url https://download.pytorch.org/whl/cu118
Error code: 2
Press any key to continue . . .>

**Comment by lamachine** [comment]
I found out my video card (nVidia 1650 does not actually support CUDA and the installer does not tell you that.  I bought a new card (4070) and installed and ran EasyDiffusion.  It found and used the video card, but does not have the options I was looking for from AUTOMATIC1111,  I ran the installer again from a clean clone and got the same error.  

11:42:25-407996 INFO     nVidia CUDA toolkit detected
11:42:25-410753 INFO     Installing package: torch torchvision --index-url https://download.pytorch.org/whl/cu118
11:43:07-219917 ERROR    Error running pip: install --upgrade torch torchvision --index-url
                         https://download.pytorch.org/whl/cu118
11:43:07-993385 ERROR    Could not load torch: DLL load failed while importing _C: The specified module could not be
                         found.

I still have the necessary packages installed already in python 3.10
torch==2.0.0+cu118
torchdiffeq==0.2.3
torchmetrics==0.11.4
torchsde==0.2.5
torchvision==0.15.1+cu118
